mm2src/mm2_main/src/lp_wallet.rs

```diff
@@ -0,0 +1,504 @@
+///   "encrypted_mnemonic_data": {
+///     // EncryptedData fields go here
+///   }
+/// }
+/// ```
+///
+/// For a `GetMnemonicResponse` where the `MnemonicForRpc` is `PlainText` with a mnemonic of "your_mnemonic_here", the JSON representation would be:
+/// ```json
+/// {
+///   "format": "plaintext",
+///   "mnemonic": "your_mnemonic_here"
+/// }
+/// ```
+#[derive(Serialize)]
+pub struct GetMnemonicResponse {
+    #[serde(flatten)]
+    pub mnemonic: MnemonicForRpc,
+}
+
+#[derive(Debug, Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum GetMnemonicError {
+    #[display(fmt = "Invalid request error: {}", _0)]
+    InvalidRequest(String),
+    #[display(fmt = "Wallets storage error: {}", _0)]
+    WalletsStorageError(String),
+    #[display(fmt = "Internal error: {}", _0)]
+    Internal(String),
+}
+
+impl HttpStatusCode for GetMnemonicError {
+    fn status_code(&self) -> StatusCode {
+        match self {
+            GetMnemonicError::InvalidRequest(_) => StatusCode::BAD_REQUEST,
+            GetMnemonicError::WalletsStorageError(_) | GetMnemonicError::Internal(_) => {
+                StatusCode::INTERNAL_SERVER_ERROR
+            },
+        }
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl From<WalletsStorageError> for GetMnemonicError {
+    fn from(e: WalletsStorageError) -> Self { GetMnemonicError::WalletsStorageError(e.to_string()) }
+}
+
+#[cfg(target_arch = "wasm32")]
+impl From<WalletsDBError> for GetMnemonicError {
+    fn from(e: WalletsDBError) -> Self { GetMnemonicError::WalletsStorageError(e.to_string()) }
+}
+
+impl From<ReadPassphraseError> for GetMnemonicError {
+    fn from(e: ReadPassphraseError) -> Self { GetMnemonicError::WalletsStorageError(e.to_string()) }
+}
+
+/// Retrieves the wallet mnemonic in the requested format.
+///
+/// # Returns
+///
+/// A `Result` type containing:
+///
+/// * [`Ok`]([`GetMnemonicResponse`]) - The wallet mnemonic in the requested format.
+/// * [`MmError`]<[`GetMnemonicError>`]> - Returns specific [`GetMnemonicError`] variants for different failure scenarios.
+///
+/// # Errors
+///
+/// This function will return an error in the following situations:
+///
+/// * The wallet name is not found in the context.
+/// * The wallet is initialized without a name.
+/// * The wallet passphrase file is not found for `MnemonicFormat::Encrypted`.
+/// * The wallet mnemonic file is not found for `MnemonicFormat::PlainText`.
+///
+/// # Examples
+///
+/// ```rust
+/// let ctx = MmArc::new(MmCtx::default());
+/// let req = GetMnemonicRequest {
+///     mnemonic_format: MnemonicFormat::Encrypted,
+/// };
+/// let result = get_mnemonic_rpc(ctx, req).await;
+/// match result {
+///     Ok(response) => println!("Mnemonic: {:?}", response.mnemonic),
+///     Err(e) => println!("Error: {:?}", e),
+/// }
+/// ```
+pub async fn get_mnemonic_rpc(ctx: MmArc, req: GetMnemonicRequest) -> MmResult<GetMnemonicResponse, GetMnemonicError> {
```

Comment:
I wonder maybe we could have a special rpc to return if mnemonic file was created. So GUI can ensure that migration to the api mnemonic has already happened and begin to use it

Reply:
It's not need IMHO, using the new `wallet_name` config field ensures that https://github.com/KomodoPlatform/komodo-defi-framework/blob/4f666c13d664488fba529e44e6d1b3391e5da030/mm2src/mm2_main/src/lp_wallet.rs#L159-L162 if not used, then it will be the same behaviour as before this PR. GUIs currently load the seed file and decrypt it then pass the mnemonic to mm2 initialization parameters. To migrate, they would pass the mnemonic with the password and the wallet name.